docs/platform-engineers/workflow/working-mechanism.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Working Mechanism
+---
+
+This document will give a brief introduction to the core mechanisms of KubeVela Workflow.
+
+## Running Mode
+
+The execution of workflow has two different running modes: **DAG** mode and **StepByStep** mode. In DAG mode, all steps in the workflow will execute concurrently. They will form a dependency graph for running according to the Input/Output in the step configuration automatically. If one workflow step has not met all its dependencies, it will wait for the conditions. In StepByStep mode, all steps will be executed in order. In KubeVela v1.2+, the defaut running mode is StepByStep. Currently, we do not support using DAG mode explicitly.
+
```

Comment:
please add some examples about when it's runing in step by step and when it will run in DAG mode.

Reply:
Instead of giving examples, I would prefer not to talk too much about the running mode in details now. Our workflow running mode is still in development and the design is not clear.

Reply:
The examples could be weird and confusing. For example, user may ask how to use DAG mode when using workflow explicitly like the implicit one and our answer is: no, you cannot. So this document is more of explaining the underlying mechanism rather than guiding user how to use it. Though it is still true that examples can make the introduction more clear.